guiunit-example: Failed to create Test.app bundle

Hi, 
For me this sample doesn't work. I'am trying to run it on macOS Mojave. 
The compilations seems to work but when trying to create the Test.app bundle, it fails.

Run: make run 

Output:
guiunit-example ciprian-ionut.holbia$ make run
mkdir obj
if [ ! -d "obj/guiunit" ]; then git clone https://github.com/mono/guiunit.git obj/guiunit; fi
Cloning into 'obj/guiunit'...
remote: Counting objects: 1084, done.
remote: Total 1084 (delta 0), reused 0 (delta 0), pack-reused 1084
Receiving objects: 100% (1084/1084), 1.74 MiB | 1.47 MiB/s, done.
Resolving deltas: 100% (647/647), done.
xbuild obj/guiunit/src/framework/GuiUnit_NET_4_5.csproj 

>>>> xbuild tool is deprecated and will be removed in future updates, use msbuild instead <<<<

XBuild Engine Version 14.0
Mono, Version 5.12.0.0
Copyright (C) 2005-2013 Various Mono authors

Build started 10/8/2018 12:58:56 PM.
__________________________________________________
Project "/Users/ciprian-ionut.holbia/Downloads/guiunit-example/obj/guiunit/src/framework/GuiUnit_NET_4_5.csproj" (default target(s)):
	Target PrepareForBuild:
		Configuration: Debug Platform: AnyCPU
		Created directory "../../bin/net_4_5/"
		Created directory "../../bin/Debug/net-4.5/"
	Target GenerateSatelliteAssemblies:
	No input files were specified for target GenerateSatelliteAssemblies, skipping.
	Target CoreCompile:
		Tool /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5/csc.exe execution started with arguments: /noconfig /debug:full /debug:portable /keyfile:guiunit.snk /publicsign /optimize- /out:obj/Debug/net-4.5/GuiUnit.exe Api/ExpectedExceptionData.cs Api/FailureSite.cs Api/IParameterDataSource.cs Api/IPropertyBag.cs Api/ITest.cs Api/ITestAssemblyBuilder.cs Api/ITestAssemblyRunner.cs Api/ITestCaseData.cs Api/ITestCaseSource.cs Api/ITestFilter.cs Api/ITestListener.cs Api/ITestResult.cs Api/IXmlNodeBuilder.cs Api/PropertyEntry.cs Api/ResultState.cs Api/RunState.cs Api/TestOutput.cs Api/TestStatus.cs Api/XmlNode.cs AssemblyInfo.cs Assert.cs AssertionHelper.cs Assume.cs Attributes/CategoryAttribute.cs Attributes/CombinatorialAttribute.cs Attributes/CultureAttribute.cs Attributes/DataAttribute.cs Attributes/DatapointAttribute.cs Attributes/DatapointsAttribute.cs Attributes/DatapointSourceAttribute.cs Attributes/DescriptionAttribute.cs Attributes/ExpectedExceptionAttribute.cs Attributes/ExplicitAttribute.cs Attributes/IgnoreAttribute.cs Attributes/IncludeExcludeAttribute.cs Attributes/MaxTimeAttribute.cs Attributes/NUnitAttribute.cs Attributes/PairwiseAttribute.cs Attributes/PlatformAttribute.cs Attributes/PropertyAttribute.cs Attributes/RandomAttribute.cs Attributes/RangeAttribute.cs Attributes/RepeatAttribute.cs Attributes/SequentialAttribute.cs Attributes/SetCultureAttribute.cs Attributes/SetUICultureAttribute.cs Attributes/SetUpAttribute.cs Attributes/TearDownAttribute.cs Attributes/TestAttribute.cs Attributes/TestCaseAttribute.cs Attributes/TestCaseSourceAttribute.cs Attributes/TestFixtureAttribute.cs Attributes/TestFixtureSetUpAttribute.cs Attributes/TestFixtureTearDownAttribute.cs Attributes/TheoryAttribute.cs Attributes/TimeoutAttribute.cs Attributes/ValuesAttribute.cs Attributes/ValueSourceAttribute.cs Constraints/AllItemsConstraint.cs Constraints/AndConstraint.cs Constraints/AssignableFromConstraint.cs Constraints/AssignableToConstraint.cs Constraints/AttributeConstraint.cs Constraints/AttributeExistsConstraint.cs Constraints/BasicConstraint.cs Constraints/BinaryConstraint.cs Constraints/BinarySerializableConstraint.cs Constraints/CollectionConstraint.cs Constraints/CollectionContainsConstraint.cs Constraints/CollectionEquivalentConstraint.cs Constraints/CollectionItemsEqualConstraint.cs Constraints/CollectionOrderedConstraint.cs Constraints/CollectionSubsetConstraint.cs Constraints/CollectionTally.cs Constraints/ComparisonAdapter.cs Constraints/ComparisonConstraint.cs Constraints/Constraint.cs Constraints/ConstraintBuilder.cs Constraints/ConstraintExpression.cs Constraints/ConstraintExpressionBase.cs Constraints/ConstraintFactory.cs Constraints/ContainsConstraint.cs Constraints/DelayedConstraint.cs Constraints/EmptyCollectionConstraint.cs Constraints/EmptyConstraint.cs Constraints/EmptyDirectoryConstraint.cs Constraints/EmptyStringConstraint.cs Constraints/EndsWithConstraint.cs Constraints/EqualConstraint.cs Constraints/EqualityAdapter.cs Constraints/ExactCountConstraint.cs Constraints/ExactTypeConstraint.cs Constraints/ExceptionTypeConstraint.cs Constraints/FailurePoint.cs Constraints/FalseConstraint.cs Constraints/FloatingPointNumerics.cs Constraints/GreaterThanConstraint.cs Constraints/GreaterThanOrEqualConstraint.cs Constraints/InstanceOfTypeConstraint.cs Constraints/IResolveConstraint.cs Constraints/LessThanConstraint.cs Constraints/LessThanOrEqualConstraint.cs Constraints/MessageWriter.cs Constraints/MsgUtils.cs Constraints/NaNConstraint.cs Constraints/NoItemConstraint.cs Constraints/NotConstraint.cs Constraints/NullConstraint.cs Constraints/NullOrEmptyStringConstraint.cs Constraints/Numerics.cs Constraints/NUnitComparer.cs Constraints/NUnitEqualityComparer.cs Constraints/Operators/AllOperator.cs Constraints/Operators/AndOperator.cs Constraints/Operators/AttributeOperator.cs Constraints/Operators/BinaryOperator.cs Constraints/Operators/CollectionOperator.cs Constraints/Operators/ConstraintOperator.cs Constraints/Operators/ExactCountOperator.cs Constraints/Operators/NoneOperator.cs Constraints/Operators/NotOperator.cs Constraints/Operators/OrOperator.cs Constraints/Operators/PrefixOperator.cs Constraints/Operators/PropOperator.cs Constraints/Operators/SelfResolvingOperator.cs Constraints/Operators/SomeOperator.cs Constraints/Operators/ThrowsOperator.cs Constraints/Operators/WithOperator.cs Constraints/OrConstraint.cs Constraints/PathConstraint.cs Constraints/PredicateConstraint.cs Constraints/PrefixConstraint.cs Constraints/PropertyConstraint.cs Constraints/PropertyExistsConstraint.cs Constraints/RangeConstraint.cs Constraints/RegexConstraint.cs Constraints/ResolvableConstraintExpression.cs Constraints/ReusableConstraint.cs Constraints/SameAsConstraint.cs Constraints/SamePathConstraint.cs Constraints/SamePathOrUnderConstraint.cs Constraints/SomeItemsConstraint.cs Constraints/StartsWithConstraint.cs Constraints/StringConstraint.cs Constraints/SubPathConstraint.cs Constraints/SubstringConstraint.cs Constraints/ThrowsConstraint.cs Constraints/ThrowsNothingConstraint.cs Constraints/Tolerance.cs Constraints/ToleranceMode.cs Constraints/TrueConstraint.cs Constraints/TypeConstraint.cs Constraints/UniqueItemsConstraint.cs Constraints/XmlSerializableConstraint.cs Contains.cs Env.cs Exceptions/AssertionException.cs Exceptions/IgnoreException.cs Exceptions/InconclusiveException.cs Exceptions/SuccessException.cs Extensibility/IParameterDataProvider.cs Extensibility/ISuiteBuilder.cs Extensibility/ITestCaseBuilder.cs Extensibility/ITestCaseProvider.cs GlobalSettings.cs Guard.cs Has.cs IExpectException.cs Internal/AssemblyHelper.cs Internal/Builders/CombinatorialStrategy.cs Internal/Builders/CombinatorialTestCaseProvider.cs Internal/Builders/CombiningStrategy.cs Internal/Builders/DataAttributeTestCaseProvider.cs Internal/Builders/DatapointProvider.cs Internal/Builders/NUnitTestCaseBuilder.cs Internal/Builders/NUnitTestFixtureBuilder.cs Internal/Builders/PairwiseStrategy.cs Internal/Builders/ParameterDataProvider.cs Internal/Builders/ProviderCache.cs Internal/Builders/SequentialStrategy.cs Internal/Commands/ApplyChangesToContextCommand.cs Internal/AsyncInvocationRegion.cs Internal/AsyncSynchronizationContext.cs Internal/Commands/CommandDecoratorList.cs Internal/Commands/CommandStage.cs Internal/Commands/DelegatingTestCommand.cs Internal/Commands/ExpectedExceptionCommand.cs Internal/Commands/ICommandDecorator.cs Internal/Commands/MaxTimeCommand.cs Internal/Commands/OneTimeSetUpCommand.cs Internal/Commands/OneTimeTearDownCommand.cs Internal/Commands/RepeatedTestCommand.cs Internal/Commands/SetUpTearDownCommand.cs Internal/Commands/SkipCommand.cs Internal/Commands/TestCommand.cs Internal/Commands/TestMethodCommand.cs Internal/Commands/TheoryResultCommand.cs Internal/CultureDetector.cs Internal/ExceptionHelper.cs Internal/Extensibility/ParameterDataProviders.cs Internal/Extensibility/TestCaseProviders.cs Internal/Filters/AndFilter.cs Internal/Filters/CategoryExpression.cs Internal/Filters/CategoryFilter.cs Internal/Filters/NotFilter.cs Internal/Filters/OrFilter.cs Internal/Filters/SimpleCategoryExpression.cs Internal/Filters/SimpleNameFilter.cs Internal/IApplyToContext.cs Internal/IApplyToTest.cs Internal/InvalidTestFixtureException.cs Internal/MethodHelper.cs Internal/NUnitException.cs Internal/NUnitLiteTestAssemblyBuilder.cs Internal/NUnitLiteTestAssemblyRunner.cs Internal/OSPlatform.cs Internal/ParameterSet.cs Internal/PlatformHelper.cs Internal/PropertyBag.cs Internal/PropertyNames.cs Internal/RandomGenerator.cs Internal/Randomizer.cs Internal/Reflect.cs Internal/Results/TestCaseResult.cs Internal/Results/TestResult.cs Internal/Results/TestSuiteResult.cs Internal/RuntimeFramework.cs Internal/StackFilter.cs Internal/StringUtil.cs Internal/TestExecutionContext.cs Internal/TestFilter.cs Internal/TestFixtureBuilder.cs Internal/TestListener.cs Internal/Tests/ParameterizedFixtureSuite.cs Internal/Tests/ParameterizedMethodSuite.cs Internal/Tests/Test.cs Internal/Tests/TestAssembly.cs Internal/Tests/TestFixture.cs Internal/Tests/TestMethod.cs Internal/Tests/TestSuite.cs Internal/TextMessageWriter.cs Internal/ThreadUtility.cs Internal/TypeHelper.cs Internal/WorkItems/CompositeWorkItem.cs Internal/WorkItems/SimpleWorkItem.cs Internal/WorkItems/WorkItem.cs Internal/WorkItems/WorkItemState.cs Is.cs Iz.cs ListMapper.cs MessageMatch.cs ObjectList.cs Runner/CommandLineOptions.cs Runner/ConsoleWriter.cs Runner/DebugWriter.cs Runner/OutputWriters/NUnit2XmlOutputWriter.cs Runner/OutputWriters/NUnit3XmlOutputWriter.cs Runner/OutputWriters/OutputWriter.cs Runner/ResultReporter.cs Runner/ResultSummary.cs Runner/TcpWriter.cs SpecialValue.cs TestCaseData.cs TestContext.cs Throws.cs GuiUnit/TestRunner.cs GuiUnit/MainLoopCommand.cs GuiUnit/IMainLoopIntegration.cs GuiUnit/XwtMainLoopIntegration.cs GuiUnit/InvokerHelper.cs GuiUnit/MonoMacMainLoopIntegration.cs GuiUnit/XmlTestListener.cs GuiUnit/AdditionalAsserts.cs GuiUnit/GtkMainLoopIntegration.cs Constraints/CollectionSupersetConstraint.cs obj/Debug/net-4.5/.NETFramework,Version=v4.5.AssemblyAttribute.cs /target:exe /define:"TRACE;DEBUG;NET_4_5;CLR_4_0,NUNITLITE" /nowarn:1574;1591 /doc:../../bin/Debug/net-4.5/nunitlite.xml /nostdlib /reference:/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5-api/System.dll /reference:/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5-api/System.Xml.dll /reference:/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5-api/System.Xml.Linq.dll /reference:/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5-api/System.Core.dll /reference:/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5-api//mscorlib.dll /warn:4
		Microsoft (R) Visual C# Compiler version 2.6.0.62309 (d3f6b8e7)
		Copyright (C) Microsoft Corporation. All rights reserved.
	Target DeployOutputFiles:
		Copying file from '/Users/ciprian-ionut.holbia/Downloads/guiunit-example/obj/guiunit/src/framework/obj/Debug/net-4.5/GuiUnit.pdb' to '/Users/ciprian-ionut.holbia/Downloads/guiunit-example/obj/guiunit/bin/net_4_5/GuiUnit.pdb'
		Copying file from '/Users/ciprian-ionut.holbia/Downloads/guiunit-example/obj/guiunit/src/framework/obj/Debug/net-4.5/GuiUnit.exe' to '/Users/ciprian-ionut.holbia/Downloads/guiunit-example/obj/guiunit/bin/net_4_5/GuiUnit.exe'
Done building project "/Users/ciprian-ionut.holbia/Downloads/guiunit-example/obj/guiunit/src/framework/GuiUnit_NET_4_5.csproj".

Build succeeded.
	 0 Warning(s)
	 0 Error(s)

Time Elapsed 00:00:01.4684140
cp obj/guiunit/bin/net_4_5/GuiUnit.exe obj/
mcs -out:obj/Test.dll -t:library -debug -r:obj/GuiUnit.exe -r:/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Xamarin.Mac.dll Test.cs
/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/bin/mmp --output obj/Test.app obj/GuiUnit.exe -a /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac.dll --nolink --profile mobile --arch i386
error MM0000: Unexpected error - Please file a bug report at http://bugzilla.xamarin.com
System.ArgumentNullException: Value cannot be null.
Parameter name: path1
  at System.IO.Path.Combine (System.String path1, System.String path2, System.String path3, System.String path4) [0x00038] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.IO/Path.cs:867 
  at Xamarin.Bundler.Driver.ValidateXcode () [0x00014] in /Users/builder/jenkins/workspace/xamarin-macios/xamarin-macios/tools/mmp/driver.cs:592 
  at Xamarin.Bundler.Driver.Main2 (System.String[] args) [0x00f87] in /Users/builder/jenkins/workspace/xamarin-macios/xamarin-macios/tools/mmp/driver.cs:480 
  at Xamarin.Bundler.Driver.Main (System.String[] args) [0x00015] in /Users/builder/jenkins/workspace/xamarin-macios/xamarin-macios/tools/mmp/driver.cs:211 
make: *** [obj/Test.app] Error 1

--- a/guiunit-example/Makefile
+++ b/guiunit-example/Makefile
@@ -5,7 +5,7 @@ clean::
 	rm -rf ./obj/
 
 obj/Test.app/: obj/ obj/Test.dll obj/GuiUnit.exe
-	/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/bin/mmp --output obj/Test.app obj/GuiUnit.exe -a /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Xamarin.Mac.dll --nolink --profile mobile --arch i386
+	/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/bin/mmp --output obj/Test.app obj/GuiUnit.exe -a /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Xamarin.Mac.dll --nolink --profile mobile --arch x86_64
 
 obj/:
 	mkdir obj
@@ -17,5 +17,5 @@ obj/guiunit/: obj/
 	if [ ! -d "obj/guiunit" ]; then git clone https://github.com/mono/guiunit.git obj/guiunit; fi
 
 obj/GuiUnit.exe: obj/ obj/guiunit/
-	xbuild obj/guiunit/src/framework/GuiUnit_NET_4_5.csproj 
+	msbuild obj/guiunit/src/framework/GuiUnit_NET_4_5.csproj
 	cp obj/guiunit/bin/net_4_5/GuiUnit.exe obj/

